### PR TITLE
LPS-173010 Import/Export picklist

### PR DESCRIPTION
**Pull request by murilostodolni**
Related: [LPS-167536](https://issues.liferay.com/browse/LPS-167536)

cc: @alinedoleron

### CODE REVIEW

**Comment by murilostodolni** [comment]
Force push to organize commits.

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
Force push to solve bug when add new object field as state.

**Comment by murilostodolni** [comment]
ci:test:sf

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts [hunk 1]
```diff
@@ -249,9 +249,9 @@ export async function getPickList(pickListId: number): Promise<PickList> {
 	);
 }
 
-export async function getPickListItems(pickListId: number) {
+export async function getPickListItems(externalReferenceCode: string) {
```
Comment (rebsilva) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts]: Was this change intentional? I don't remember changing this part when I did the commit, so maybe the branch wasn't updated when was pushed causing this? Can you please check if this should be pickListId or externalReferenceCode?

Also please check the rest of this commit if things as they should be comparing with your local, thanks!

**Review by rebsilva — CHANGES_REQUESTED** [review]
Hi @murilostodolni and @alinedoleron! Great job on this, I'm just wondering if one of the commits made some changes that were not necessary, could you please check my comment? If everything is okay please let me know that I will keep reviewing the PR, thanks! :purple_heart:

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @murilostodolni, great work!
I finished my review. Let me know if you have questions :smile:

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -171,6 +183,19 @@ private ListTypeDefinition _addListTypeDefinition(
 		return listTypeDefinition;
 	}
 
+	private void _addListTypeEntries(
+			long listTypeDefinitionId, List<ListTypeEntry> listTypeEntries,
+			long userId)
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:188]: Done!

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -79,6 +79,16 @@ public ListTypeEntry addListTypeEntry(
 		return listTypeEntryPersistence.update(listTypeEntry);
 	}
 
+	@Override
+	public void deleteByListTypeDefinitionId(long listTypeDefinitionId) {
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java:83]: Done

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Review by rebsilva — APPROVED** [review]
LGTM! Great work @alinedoleron! :purple_circle: :cherry_blossom:

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
Force push to rebase with brianchandtocom's master

**Comment by murilostodolni** [comment]
Forwarding manually: https://github.com/brianchandotcom/liferay-portal/pull/129695

**Comment by murilostodolni** [comment]
ci:close

**Pull request closed without merging**